src/FsAutoComplete.Core/Utils.fs

```diff
@@ -728,28 +728,49 @@ module Version =
       GitSha = sha }
 
 //source: https://nbevans.wordpress.com/2014/08/09/a-simple-stereotypical-javascript-like-debounce-service-for-f/
-type Debounce<'a>(timeout, fn) =
-  let debounce fn timeout =
+type Debounce<'a>(timeout, fn) as x =
```

Comment:
curious about the `as x` pattern here - I see it used below in the `mailbox`, is that required because `mailbox` became a value?

Reply:
`MyType(...) as x` -> binds `x` as `this` identifier. Basically just like `member x.MyMethod(...)=..`, but here `x` is just valid for current method, while `x` in `MyType(...) as x` is valid for the whole class -- including outside of members (like inside let bindings).  

`let mailbox = ...` isn't a member or inside a member, but [calls a class member](https://github.com/Booksbaum/FsAutoComplete/blob/07629d1025541490ed3dd7f037eb4ec5f02cdb6b/src/FsAutoComplete.Core/Utils.fs#L743) -> needs access to `this` (here named `x`). Wasn't necessary previously because no member was called (-> no access to `this` was necessary).

Reply:
ok, so if we make `WaitForFnToFinish` the default (as I believe it should be), then the 'self' identifier could be removed entirely.

Reply:
Yes

I just introduced the property `WaitForFnToFinish` for testing -> to change on the fly how it should behave (`FSharp.debug.waitTillFileChecked`)